chirps/scan/tasks.py

```diff
@@ -75,37 +77,14 @@ def scan_task(scan_asset_id):
     total_rules = len(policy_rules)
     rules_run = 0
     for policy, rule in policy_rules:
-        logger.info('Starting rule evaluation', extra={'id': rule.id})
-
-        if asset.REQUIRES_EMBEDDINGS:
-            # template policies should not be bound to a user
-            add_user_filter = not policy.is_template
-            user = scan_run.scan_version.scan.user if add_user_filter else None
-            embedding = create_embedding(rule.query_string, asset.embedding_model, asset.embedding_model_service, user)
-            query = embedding.vectors
-        else:
-            query = rule.query_string
-
-        # Issue the search query to the asset provider
-        results: list[SearchResult] = asset.search(query, max_results=100)
-
-        for search_result in results:
-
-            # Create the result. We'll flip the result flag to True if any findings are found
-            result = RegexResult(rule=rule, text=search_result.data, scan_asset=scan_asset)
-            result.save()
 
-            # Run the regex against the text
-            for match in re.finditer(rule.regex_test, search_result.data):
-
-                # Persist the finding
-                finding = RegexFinding(
-                    result=result,
-                    offset=match.start(),
-                    length=match.end() - match.start(),
-                    source_id=search_result.source_id,
-                )
-                finding.save()
+        if type(rule) is RegexRule:
+            _execute_regex_rule(rule=rule, scan_asset=scan_asset, asset=asset)
```

Comment:
instead of this, what if we add an execute method to the BaseRule and override it in RegexRule so that we don't even need to check the type?